File: tmdb-core/src/main/java/dev/reuss/tmdb/query/AppendableResponse.java
package dev.reuss.tmdb.query;

import java.io.Serializable;

/**
 * Common contract for append_to_response values.
 */
public interface AppendableResponse extends Serializable {

    /**
     * TMDB append_to_response query value.
     *
     * @return append value
     */
    String value();

    /**
     * Response model type for this appended response.
     *
     * @return response type
     */
    Class<?> responseType();
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/query/AppendableResponse.kt
package dev.reuss.tmdb.query

import java.io.Serializable

/**
 * Common contract for TMDB `append_to_response` values.
 *
 * Implementations are intended for endpoint groups that support appended
 * responses and expose both the raw TMDB query value and the expected response
 * model type.
 */
interface AppendableResponse : Serializable {
    /**
     * TMDB `append_to_response` query value.
     */
    val value: String

    /**
     * Response model type for this appended response.
     */
    val responseType: Class<*>
}
